src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] hashTable;
+
+    public MyHashMap() {
+        hashTable = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= getThreshold()) {
+            resize();
+        }
+        int index = hashKey(key);
+        Node<K, V> node = hashTable[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        addNode(key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        int index = hashKey(key);
+        Node<K, V> node = hashTable[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+
+        return size;
+    }
+
+    private float getThreshold() {
+        float threshold = hashTable.length * LOAD_FACTOR;
+        return threshold;
+    }
+
+    private void resize() {
+        size = 0;
+        Node<K, V>[] oldTable = hashTable;
+        hashTable = new Node[oldTable.length * 2];
```

Comment:
Use the diamond operator to avoid raw type usage. It should be 'new Node<>[oldTable.length * 2]' to specify that you are creating an array for a generic Node class without explicit type parameters.